ui/src/views/storage/CreateFileShare.vue

```diff
@@ -0,0 +1,492 @@
+        serviceofferingid: [{ required: true, message: this.$t('label.required') }],
+        diskofferingid: [{ required: true, message: this.$t('label.required') }],
+        size: [{ required: true, message: this.$t('message.error.custom.disk.size') }],
+        miniops: [{
+          validator: async (rule, value) => {
+            if (value && (isNaN(value) || value <= 0)) {
+              return Promise.reject(this.$t('message.error.number'))
+            }
+            return Promise.resolve()
+          }
+        }],
+        maxiops: [{
+          validator: async (rule, value) => {
+            if (value && (isNaN(value) || value <= 0)) {
+              return Promise.reject(this.$t('message.error.number'))
+            }
+            return Promise.resolve()
+          }
+        }]
+      })
+    },
+    arrayHasItems (array) {
+      return array !== null && array !== undefined && Array.isArray(array) && array.length > 0
+    },
+    fetchOwnerOptions (OwnerOptions) {
+      this.owner = {}
+      console.log('fetching owner')
+      if (OwnerOptions.selectedAccountType === this.$t('label.account')) {
+        if (!OwnerOptions.selectedAccount) {
+          return
+        }
+        console.log('fetched account')
+        this.owner.account = OwnerOptions.selectedAccount
+        this.owner.domainid = OwnerOptions.selectedDomain
+      } else if (OwnerOptions.selectedAccountType === this.$t('label.project')) {
+        if (!OwnerOptions.selectedProject) {
+          return
+        }
+        console.log('fetched project')
+        this.owner.projectid = OwnerOptions.selectedProject
+      }
+      console.log('fetched owner')
+      this.fetchData()
+    },
+    fetchData () {
+      this.fetchZones()
+      this.fetchFileShareProviders()
+    },
+    fetchZones () {
+      this.zoneLoading = true
+      const params = { showicon: true }
+      api('listZones', params).then(json => {
+        var listZones = json.listzonesresponse.zone
+        if (listZones) {
+          listZones = listZones.filter(x => x.allocationstate === 'Enabled')
+          this.zones = this.zones.concat(listZones)
+        }
+      }).finally(() => {
+        this.zoneLoading = false
+        if (this.arrayHasItems(this.zones)) {
+          this.form.zoneid = this.zones[0].id
+          this.handleZoneChange(this.zones[0])
+        }
+      })
+    },
+    handleZoneChange (zone) {
+      this.selectedZone = zone
+      this.fetchServiceOfferings()
+      this.fetchDiskOfferings()
+      this.fetchNetworks()
+    },
+    fetchFileShareProviders (id) {
+      this.providersLoading = true
+      api('listFileShareProviders').then(json => {
+        this.providers = json.listfileshareprovidersresponse.fileshareprovider || []
+        this.form.provider = this.providers[0].name || ''
+      }).finally(() => {
+        this.providersLoading = false
+      })
+    },
+    fetchCapabilities (id) {
```

Comment:
we already do listCapabilities call during user login, I guess we can use that
Search for `$store.getters.features.` in UI code

Reply:
thanks, done